model/src/main/kotlin/OrtResult.kt

```diff
@@ -311,7 +305,33 @@ data class OrtResult(
      * Return true if the project or package with the given identifier is excluded.
      */
     @Suppress("UNUSED") // This is intended to be mostly used via scripting.
-    fun isExcluded(id: Identifier): Boolean = getExcludes().isExcluded(id, this)
+    fun isExcluded(id: Identifier): Boolean =
+        getProjects().find { it.id == id }?.let {
+            // An excluded project could still be included as a dependency of another non-excluded project.
+            isProjectExcluded(it) && isPackageExcluded(id)
+        } ?: isPackageExcluded(id)
+
+
+    /**
+     * True if all occurrences of the package identified by [id] in the [ortResult] are excluded by this [Excludes]
+     * configuration.
+     */
+    fun isPackageExcluded(id: Identifier): Boolean =
+        getProjects().all { project ->
+            isProjectExcluded(project) || project.scopes.all { scope ->
+                getExcludes().isScopeExcluded(scope, project, this) || !scope.contains(id)
+            }
+        }
+
+    /**
+     * True if the [project] is excluded by this [Excludes] configuration.
```

Comment:
I would remove "by this [Excludes] configuration", it doesn't add valuable information and it's confusing what "this" refers to. Or if you want to mention it say something like "by the [Excludes] defined in the [RepositoryConfiguration] of [repository]."